extensions/jacksonxml/deployment/src/main/java/org/apache/camel/quarkus/component/jacksonxml/deployment/JacksonxmlProcessor.java

```diff
@@ -28,4 +37,21 @@ FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
+    @BuildStep
+    ReflectiveClassBuildItem registerJsonView(CombinedIndexBuildItem combinedIndex) {
+
+        IndexView index = combinedIndex.getIndex();
+        DotName JSON_VIEW = DotName.createSimple(JsonView.class.getName());
+        String[] jsonViews = index.getAnnotations(JSON_VIEW).stream().map(ai -> ai.value())
+                .filter(p -> AnnotationValue.Kind.ARRAY.equals(p.kind()))
+                .map((Function<? super AnnotationValue, ? extends String>) ai -> {
+                    Type[] annotationType = ai.asClassArray();
+                    return annotationType[0].name().toString();
```

Comment:
Why are we taking only the first array element into account? IMO all elements need to be registered for reflection.

Reply:
You are right, I will send another commit.
Thanks!

Reply:
Hi @ppalaga ,

And here is the follow up PR
https://github.com/apache/camel-quarkus/pull/2709

Please take a look.
Freeman